addons/mail/models/mail_user_settings.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models
+
+
+class MailUserSettings(models.Model):
+    _name = 'mail.user.settings'
+    _description = 'Mail User Settings'
+
+    is_discuss_sidebar_category_channel_open = fields.Boolean(string="Is discuss sidebar category channel open?", default=True)
+    is_discuss_sidebar_category_chat_open = fields.Boolean(string="Is discuss sidebar category chat open?", default=True)
+    user_id = fields.Many2one('res.users', string="User", required=True, index=True, ondelete='cascade')
+
+    # Rtc
+    push_to_talk_key = fields.Char()
+    use_push_to_talk = fields.Boolean(default=False)
+    voice_active_duration = fields.Integer()
+    volume_settings_ids = fields.One2many('mail.volume.setting', 'user_setting_id')
+
+    _sql_constraints = [
+        ('unique_user_id', 'UNIQUE(user_id)', 'One user should only have one mail user settings.')
+    ]
+
+    @api.model
+    def find_or_create_for_user(self, user):
+        settings = user.mail_user_settings
+        if not settings:
+            settings = self.create({'user_id': user.id})
+        return settings
+
+    def _mail_user_settings_format_fields(self):
+        return [
+            'id',
+            'is_discuss_sidebar_category_channel_open',
+            'is_discuss_sidebar_category_chat_open',
+            'user_id',
+            'push_to_talk_key',
+            'voice_active_duration',
+            'use_push_to_talk',
+        ]
+
+    def mail_user_settings_format(self):
+        self.ensure_one()
+        res = self._read_format(fnames=self._mail_user_settings_format_fields())[0]
```

Comment:
It's made for that purpose, it's just a smarter version of read that doesn't do a query if it's not necessary, to return a dict of field/value pairs the same way as read.

Reply:
That doesn't mean anything. If it actually worked in all situations `read` would just do that, so it does not, meaning it should not be used par defaults and should be applied as a justifiable and justified optimisation.

I also don't understand why `mail_user_settings_format` is accessible over RPC.

Reply:
I agree, it is was `read()` should do. I don't remember the exact reason for not changing it immediately. For more details this would need to be checked with Raphael.

But in current state `_read_format()` is just a better method, and using `read()` should actually be the case that needs a justification to explain why making a query would be desirable if the date is already in cache.

Reply:
I will make the method private as suggested though, it doesn't leak any information as public but we don't plan to call it by RPC anyway.